packages/react-components/react-overflow/src/useOverflowVisibility.ts

```diff
@@ -0,0 +1,11 @@
+import { useOverflowContext } from './overflowContext';
+
+/**
+ * A hook that returns the visibility status of all items and groups. This hook
+ * will cause the component it is in to re-render for every single time an item overflows or becomes
+ * visible - use with caution
+ * @returns visibility status of all items and groups
+ */
+export function useOverflowVisibility() {
+  return useOverflowContext(ctx => ({ itemVisibility: ctx.itemVisibility, groupVisibility: ctx.groupVisibility }));
```

Comment:
This will re-render on every context update as you return a new object reference every time.